src/app/pages/services/services.component.ts

```diff
@@ -204,48 +170,81 @@ export class ServicesComponent implements EntityTableConfig, OnInit {
         }
         service.state = ServiceStatus.Stopped;
         service.onChanging = false;
-        this.serviceTable.changeDetectorRef.detectChanges();
         waiting = false;
       }
+      this.cdr.markForCheck();
     }, (res) => {
       let message = this.translate.instant('Error starting service {serviceName}.', { serviceName });
       if (rpc === 'service.stop') {
         message = this.translate.instant('Error stopping service {serviceName}.', { serviceName });
       }
       this.dialog.errorReport(message, res.message, res.stack);
       service.onChanging = false;
-      this.serviceTable.changeDetectorRef.detectChanges();
       waiting = false;
+      this.cdr.markForCheck();
     });
   }
 
   enableToggle(service: ServiceRow): void {
-    this.ws
-      .call('service.update', [service.id, { enable: service.enable }])
+    this.services.enableDisableAction(service.id, service.enable)
       .pipe(untilDestroyed(this))
       .subscribe((res) => {
         if (!res) {
           // Middleware should return the service id
           throw new Error('Method service.update failed. No response from server');
         }
+        this.cdr.markForCheck();
       });
   }
 
-  editService(service: ServiceName): void {
-    switch (service) {
-      case ServiceName.Iscsi:
-        this.router.navigate(['/', 'sharing', 'iscsi']);
-        break;
-      case ServiceName.Cifs:
-        this.router.navigate(['/', 'services', 'smb']);
-        break;
-      default:
-        this.router.navigate(['/', 'services', service]);
-        break;
+  configureService(row: ServiceRow): void {
+    if (row.service === ServiceName.OpenVpnClient || row.service === ServiceName.OpenVpnServer) {
+      const navigationExtras: NavigationExtras = { state: { configureOpenVPN: row.service.replace('openvpn_', '') } };
+      this.router.navigate(['network'], navigationExtras);
+    } else {
+      switch (row.service) {
+        case ServiceName.Iscsi:
+          this.router.navigate(['/sharing', 'iscsi']);
+          break;
+        case ServiceName.Cifs:
+          this.router.navigate(['/services', 'smb']);
+          break;
+        default:
+          this.router.navigate(['/services', row.service]);
+          break;
+      }
     }
   }
 
-  getServiceName(service: Service): string {
-    return serviceNames.get(service.service) || service.service;
+  setupToolbar(): void {
```

Comment:
I'm assuming this will change in the future PRs.

Reply:
Yeah, probably it involves some refactoring for entity-toolbar or a new component for the page title that allows extending via templates